Index: baseclasses.vcproj
=================================================================
diff -ru8 /d/ms-sdks/Windows/v6.0/Samples/Multimedia/DirectShow/BaseClasses/baseclasses.vcproj /d/ms-sdks/DirectShow-baseclasses/baseclasses.vcproj
--- baseclasses.vcproj	Wed Jan 24 09:49:18 2007
+++ baseclasses.vcproj	Wed Oct  8 12:14:43 2008
@@ -39,17 +39,17 @@
 			/>
 			<Tool
 				Name="VCCLCompilerTool"
 				Optimization="0"
 				AdditionalIncludeDirectories="."
 				PreprocessorDefinitions="WIN32;_DEBUG;_LIB"
 				MinimalRebuild="true"
 				BasicRuntimeChecks="3"
-				RuntimeLibrary="3"
+				RuntimeLibrary="1"
 				UsePrecompiledHeader="0"
 				WarningLevel="3"
 				Detect64BitPortabilityProblems="false"
 				DebugInformationFormat="4"
 			/>
 			<Tool
 				Name="VCManagedResourceCompilerTool"
 			/>
@@ -57,17 +57,17 @@
 				Name="VCResourceCompilerTool"
 			/>
 			<Tool
 				Name="VCPreLinkEventTool"
 			/>
 			<Tool
 				Name="VCLibrarianTool"
 				AdditionalDependencies="strmiids.lib"
-				OutputFile="$(OutDir)\strmbasd.lib"
+				OutputFile="$(OutDir)\strmbase.lib"
 				AdditionalLibraryDirectories=""
 			/>
 			<Tool
 				Name="VCALinkTool"
 			/>
 			<Tool
 				Name="VCXDCMakeTool"
 			/>
@@ -103,17 +103,17 @@
 			<Tool
 				Name="VCMIDLTool"
 			/>
 			<Tool
 				Name="VCCLCompilerTool"
 				Optimization="2"
 				AdditionalIncludeDirectories="."
 				PreprocessorDefinitions="WIN32;NDEBUG;_LIB;"
-				RuntimeLibrary="2"
+				RuntimeLibrary="0"
 				UsePrecompiledHeader="0"
 				WarningLevel="3"
 				Detect64BitPortabilityProblems="false"
 				DebugInformationFormat="3"
 			/>
 			<Tool
 				Name="VCManagedResourceCompilerTool"
 			/>
