--- a/config/system-headers
+++ a/config/system-headers
@@ -272,6 +272,19 @@
 freetype/ttnameid.h
 freetype/tttables.h
 freetype/t1tables.h
+freetype.h
+ftcache.h
+ftglyph.h
+ftsynth.h
+ftoutln.h
+ttnameid.h
+tttables.h
+t1tables.h
+ftlcdfil.h
+ftsizes.h
+ftadvanc.h
+ftbitmap.h
+ftxf86.h
 fribidi/fribidi.h
 FSp_fopen.h
 fstream
--- a/gfx/thebes/gfxPangoFonts.cpp
+++ a/gfx/thebes/gfxPangoFonts.cpp
@@ -73,6 +73,7 @@
 #include <cairo-ft.h>
 
 #include <fontconfig/fcfreetype.h>
+#include <freetype.h>
 #include <pango/pango.h>
 #include <pango/pangocairo.h>
 #include <pango/pango-modules.h>
--- a/js/src/config/system-headers
+++ a/js/src/config/system-headers
@@ -272,6 +272,19 @@
 freetype/ttnameid.h
 freetype/tttables.h
 freetype/t1tables.h
+freetype.h
+ftcache.h
+ftglyph.h
+ftsynth.h
+ftoutln.h
+ttnameid.h
+tttables.h
+t1tables.h
+ftlcdfil.h
+ftsizes.h
+ftadvanc.h
+ftbitmap.h
+ftxf86.h
 fribidi/fribidi.h
 FSp_fopen.h
 fstream
