--- gstreamer-0.10.17/gst/gstminiobject.c	2008-01-14 13:34:58.000000000 -0800
+++ songbird/gst-sbird/trunk/dependencies/vendor/gstreamer/gst/gstminiobject.c	2008-03-10 17:03:33.000000000 -0700
@@ -33,9 +33,9 @@
 #include "config.h"
 #endif
 
+#include "gst/gst_private.h"
 #include "gst/gstminiobject.h"
 #include "gst/gstinfo.h"
-#include "gst/gst_private.h"
 #include <gobject/gvaluecollector.h>
 
 #ifndef GST_DISABLE_TRACE
