--- gstreamer-0.10.17/common/m4/pkg.m4	2008-03-07 17:04:00.000000000 -0800
+++ songbird/gst-sbird/trunk/dependencies/vendor/gstreamer/common/m4/pkg.m4	2008-03-10 17:03:31.000000000 -0700
@@ -67,13 +67,9 @@ fi])
 # ---------------------------------------------
 m4_define([_PKG_CONFIG],
 [if test -n "$PKG_CONFIG"; then
-    if test -n "$$1"; then
-        pkg_cv_[]$1="$$1"
-    else
         PKG_CHECK_EXISTS([$3],
                          [pkg_cv_[]$1=`$PKG_CONFIG --[]$2 "$3" 2>/dev/null`],
 			 [pkg_failed=yes])
-    fi
 else
 	pkg_failed=untried
 fi[]dnl

