--- liboil-0.3.13/examples/jpeg/jpeg_debug.h	2008-02-22 11:59:17.000000000 -0800
+++ songbird/gst-sbird/trunk/dependencies/vendor/liboil/examples/jpeg/jpeg_debug.h	2008-03-11 17:10:52.000000000 -0700
@@ -2,10 +2,14 @@
 #ifndef _JPEG_DEBUG_H_
 #define _JPEG_DEBUG_H_
 
+#ifdef _MSC_VER
+#define JPEG_DEBUG(n, ...)
+#else
 #define JPEG_DEBUG(n, format...)	do{ \
 	if((n)<=JPEG_DEBUG_LEVEL)jpeg_debug((n),format); \
 }while(0)
 #define JPEG_DEBUG_LEVEL 4
+#endif
 
 void jpeg_debug(int n, const char *format, ... );
 
--- liboil-0.3.13/liboil/conv/conv_c.c	2008-02-22 11:59:17.000000000 -0800
+++ songbird/gst-sbird/trunk/dependencies/vendor/liboil/liboil/conv/conv_c.c	2008-03-12 11:16:39.000000000 -0700
@@ -33,6 +33,18 @@
 
 #include <math.h>
 
+#ifdef __MSC_VER
+#include <glib/glib.h>
+static inline gdouble
+rint(gdouble x) {
+  int xi = (int) x;
+  gdouble remainder = x - (gdouble) xi;
+
+  if (remainder < 0.5)
+    return x - remainder;
+  return (x - remainder + 1.0);
+}
+#endif
 
 
--- liboil-0.3.13/liboil/dct/dct12_f32.c	2008-02-22 11:59:17.000000000 -0800
+++ songbird/gst-sbird/trunk/dependencies/vendor/liboil/liboil/dct/dct12_f32.c	2008-03-11 17:10:51.000000000 -0700
@@ -31,6 +31,9 @@
 
 #include <liboil/liboilfunction.h>
 #include <liboil/dct/dct.h>
+#ifdef _MSC_VER
+#define _USE_MATH_DEFINES
+#endif
 #include <math.h>
 
 
diff -ur liboil-0.3.13/liboil/dct/fdct8_f64.c songbird/gst-sbird/trunk/dependencies/vendor/liboil/liboil/dct/fdct8_f64.c
--- liboil-0.3.13/liboil/dct/fdct8_f64.c	2008-02-22 11:59:17.000000000 -0800
+++ songbird/gst-sbird/trunk/dependencies/vendor/liboil/liboil/dct/fdct8_f64.c	2008-03-11 17:10:51.000000000 -0700
@@ -31,6 +31,9 @@
 
 #include <liboil/liboilfunction.h>
 #include <liboil/dct/dct.h>
+#ifdef _MSC_VER
+#define _USE_MATH_DEFINES
+#endif
 #include <math.h>
 
 /**
diff -ur liboil-0.3.13/liboil/dct/fdct8x8_f64.c songbird/gst-sbird/trunk/dependencies/vendor/liboil/liboil/dct/fdct8x8_f64.c
--- liboil-0.3.13/liboil/dct/fdct8x8_f64.c	2008-02-22 11:59:17.000000000 -0800
+++ songbird/gst-sbird/trunk/dependencies/vendor/liboil/liboil/dct/fdct8x8_f64.c	2008-03-11 17:10:51.000000000 -0700
@@ -32,6 +32,9 @@
 #include <liboil/liboilfunction.h>
 #include <liboil/liboilfuncs.h>
 #include <liboil/dct/dct.h>
+#ifdef _MSC_VER
+#define _USE_MATH_DEFINES
+#endif
 #include <math.h>
 
 
diff -ur liboil-0.3.13/liboil/dct/idct8_f64.c songbird/gst-sbird/trunk/dependencies/vendor/liboil/liboil/dct/idct8_f64.c
--- liboil-0.3.13/liboil/dct/idct8_f64.c	2008-02-22 11:59:17.000000000 -0800
+++ songbird/gst-sbird/trunk/dependencies/vendor/liboil/liboil/dct/idct8_f64.c	2008-03-11 17:10:51.000000000 -0700
@@ -31,6 +31,9 @@
 
 #include <liboil/liboilfunction.h>
 #include <liboil/dct/dct.h>
+#ifdef _MSC_VER
+#define _USE_MATH_DEFINES
+#endif
 #include <math.h>
 
 /**
diff -ur liboil-0.3.13/liboil/dct/idct8x8_c.c songbird/gst-sbird/trunk/dependencies/vendor/liboil/liboil/dct/idct8x8_c.c
--- liboil-0.3.13/liboil/dct/idct8x8_c.c	2008-02-22 11:59:17.000000000 -0800
+++ songbird/gst-sbird/trunk/dependencies/vendor/liboil/liboil/dct/idct8x8_c.c	2008-03-12 11:16:38.000000000 -0700
@@ -31,6 +31,9 @@
 
 #include <liboil/liboil.h>
 #include <liboil/dct/dct.h>
+#ifdef _MSC_VER
+#define _USE_MATH_DEFINES
+#endif
 #include <math.h>
 #include <liboil/liboiltest.h>
 #include <liboil/liboilparameter.h>
@@ -42,12 +45,19 @@
  * @short_description: DCT related functions
  */
 
+#ifdef __MSC_VER
+#define BLOCK8x8_F64(ptr, stride, row, column) \
+	(*((double *)((double *)ptr + stride*row) + column))
+
+#define BLOCK8x8_PTR_F64(ptr, stride, row, column) \
+	((double *)((double *)ptr + stride*row) + column)
+#else
 #define BLOCK8x8_F64(ptr, stride, row, column) \
 	(*((double *)((void *)ptr + stride*row) + column))
 
 #define BLOCK8x8_PTR_F64(ptr, stride, row, column) \
 	((double *)((void *)ptr + stride*row) + column)
-
+#endif
 #define BLOCK8x8_S16(ptr, stride, row, column) \
 	(*((int16_t *)((void *)ptr + stride*row) + column))
 
diff -ur liboil-0.3.13/liboil/dct/imdct32_f32.c songbird/gst-sbird/trunk/dependencies/vendor/liboil/liboil/dct/imdct32_f32.c
--- liboil-0.3.13/liboil/dct/imdct32_f32.c	2008-02-22 11:59:17.000000000 -0800
+++ songbird/gst-sbird/trunk/dependencies/vendor/liboil/liboil/dct/imdct32_f32.c	2008-03-11 17:10:51.000000000 -0700
@@ -31,6 +31,9 @@
 
 #include <liboil/liboilfunction.h>
 #include <liboil/dct/dct.h>
+#ifdef _MSC_VER
+#define _USE_MATH_DEFINES
+#endif
 #include <math.h>
 
 
--- liboil-0.3.13/liboil/deprecated/conv.c	2008-02-22 11:59:17.000000000 -0800
+++ songbird/gst-sbird/trunk/dependencies/vendor/liboil/liboil/deprecated/conv.c	2008-03-12 11:16:39.000000000 -0700
@@ -35,7 +35,17 @@
 #include <math.h>
 #include <string.h>
 #include <stdlib.h>
-
+#ifdef __MSC_VER
+static inline gdouble
+rint(gdouble x) {
+  int xi = (int) x;
+  gdouble remainder = x - (gdouble) xi;
+
+  if (remainder < 0.5)
+    return x - remainder;
+  return (x - remainder + 1.0);
+}
+#endif
 
 /**
  * SECTION:liboilfuncs-conv
--- liboil-0.3.13/liboil/fb/fbmmx.c	2008-02-22 11:59:17.000000000 -0800
+++ songbird/gst-sbird/trunk/dependencies/vendor/liboil/liboil/fb/fbmmx.c	2008-03-11 17:10:51.000000000 -0700
@@ -660,6 +660,7 @@
 #endif
 
 
+#ifdef ENABLE_BROKEN_IMPLS
 /* ------------------ MMX code paths called from fbpict.c ----------------------- */
 
 typedef union {
@@ -897,6 +898,7 @@
     
     return _mm_or_si64 (b, p);
 }
+#endif
 
 #ifdef ENABLE_BROKEN_IMPLS
 /* broken.  See Debian bug #340932 */
--- liboil-0.3.13/liboil/jpeg/convert8x8_c.c	2008-02-22 11:59:17.000000000 -0800
+++ songbird/gst-sbird/trunk/dependencies/vendor/liboil/liboil/jpeg/convert8x8_c.c	2008-03-12 11:16:38.000000000 -0700
@@ -33,6 +33,18 @@
 
 #include <math.h>
 
+#ifdef __MSC_VER
+static inline gdouble
+rint(gdouble x) {
+  int xi = (int) x;
+  gdouble remainder = x - (gdouble) xi;
+
+  if (remainder < 0.5)
+    return x - remainder;
+  return (x - remainder + 1.0);
+}
+#endif
+
 /**
  * oil_conv8x8_s16_f64:
  * @d_8x8:
@@ -73,6 +85,19 @@
 OIL_DEFINE_CLASS (clipconv8x8_u8_s16,
     "uint8_t * d_8x8, int dstr, int16_t * s_8x8, int sstr");
 
+#ifdef __MSC_VER
+#define BLOCK8x8_F64(ptr, stride, row, column) \
+	(*((double *)((double *)ptr + stride*row) + column))
+
+#define BLOCK8x8_PTR_F64(ptr, stride, row, column) \
+	((double *)((double *)ptr + stride*row) + column)
+
+#define BLOCK8x8_S16(ptr, stride, row, column) \
+	(*((int16_t *)((int16_t *)ptr + stride*row) + column))
+
+#define BLOCK8x8_U8(ptr, stride, row, column) \
+	(*((uint8_t *)((uint8_t *)ptr + stride*row) + column))
+#else
 #define BLOCK8x8_F64(ptr, stride, row, column) \
 	(*((double *)((void *)ptr + stride*row) + column))
 
@@ -84,7 +109,7 @@
 
 #define BLOCK8x8_U8(ptr, stride, row, column) \
 	(*((uint8_t *)((void *)ptr + stride*row) + column))
-
+#endif
 
 static void
 conv8x8_s16_f64_c (int16_t * dest, int dstr, double *src, int sstr)
--- liboil-0.3.13/liboil/jpeg/quantize8x8_c.c	2008-02-22 11:59:17.000000000 -0800
+++ songbird/gst-sbird/trunk/dependencies/vendor/liboil/liboil/jpeg/quantize8x8_c.c	2008-03-12 11:16:38.000000000 -0700
@@ -47,8 +47,13 @@
     "int16_t *d_8x8, int dstr, int16_t *s1_8x8, int sstr1, "
     "int16_t *s2_8x8, int sstr2");
 
+#ifdef __MSC_VER
+#define BLOCK8x8_S16(ptr, stride, row, column) \
+	(*((int16_t *)((int16_t *)ptr + stride*row) + column))
+#else
 #define BLOCK8x8_S16(ptr, stride, row, column) \
 	(*((int16_t *)((void *)ptr + stride*row) + column))
+#endif
 
 static void
 dequantize8x8_s16_ref (int16_t *dest, int dstr, int16_t *src, int sstr,
diff -ur liboil-0.3.13/liboil/jpeg/zigzag8x8_c.c songbird/gst-sbird/trunk/dependencies/vendor/liboil/liboil/jpeg/zigzag8x8_c.c
--- liboil-0.3.13/liboil/jpeg/zigzag8x8_c.c	2008-02-22 11:59:17.000000000 -0800
+++ songbird/gst-sbird/trunk/dependencies/vendor/liboil/liboil/jpeg/zigzag8x8_c.c	2008-03-12 11:16:38.000000000 -0700
@@ -59,10 +59,13 @@
 OIL_DEFINE_CLASS (unzigzag8x8_s16,
     "int16_t *d_8x8, int ds, int16_t *s_8x8, int ss");
 
-
-
+#ifdef __MSC_VER
+#define BLOCK8x8_S16(ptr, stride, row, column) \
+	(*((int16_t *)((int16_t *)ptr + stride*row) + column))
+#else
 #define BLOCK8x8_S16(ptr, stride, row, column) \
 	(*((int16_t *)((void *)ptr + stride*row) + column))
+#endif
 
 static const unsigned char zigzag_order[64] = {
 	0,
--- liboil-0.3.13/liboil/liboilcpu-x86.c	2008-02-22 11:59:17.000000000 -0800
+++ songbird/gst-sbird/trunk/dependencies/vendor/liboil/liboil/liboilcpu-x86.c	2008-03-11 17:10:52.000000000 -0700
@@ -34,14 +34,18 @@
 #include <liboil/liboilfault.h>
 #include <liboil/liboilutils.h>
 
+#ifdef HAVE_UNISTD_H
 #include <unistd.h>
+#endif
 #include <fcntl.h>
 #include <stdlib.h>
 #include <string.h>
 #include <stdio.h>
 #include <setjmp.h>
 #include <signal.h>
+#ifndef _MSC_VER
 #include <sys/time.h>
+#endif
 #include <time.h>
 
 #if defined(__FreeBSD__) || defined(__APPLE__)
@@ -129,6 +133,7 @@
 }
 #endif
 
+#ifdef HAVE_GCC_ASM
 static unsigned long
 oil_profile_stamp_rdtsc(void)
 {
@@ -286,6 +291,7 @@
   }
 }
 #endif
+#endif
 
 /* Reduce the set of CPU capabilities detected by whatever detection mechanism
  * was chosen, according to kernel limitations.  SSE requires kernel support for
@@ -328,6 +334,7 @@
 void
 oil_cpu_detect_arch(void)
 {
+#ifdef HAVE_GCC_ASM
 #ifdef USE_I386_CPUID
   oil_cpu_detect_cpuid ();
 #endif
@@ -337,6 +344,7 @@
 #ifdef USE_I386_CPUINFO
   oil_cpu_detect_cpuinfo ();
 #endif
+#endif
 
   oil_cpu_detect_kernel_support ();
 }
diff -ur liboil-0.3.13/liboil/liboilcpu.c songbird/gst-sbird/trunk/dependencies/vendor/liboil/liboil/liboilcpu.c
--- liboil-0.3.13/liboil/liboilcpu.c	2008-02-22 11:59:17.000000000 -0800
+++ songbird/gst-sbird/trunk/dependencies/vendor/liboil/liboil/liboilcpu.c	2008-03-11 17:10:52.000000000 -0700
@@ -34,14 +34,18 @@
 #include <liboil/liboilfault.h>
 #include <liboil/liboilutils.h>
 
+#ifdef HAVE_UNISTD_H
 #include <unistd.h>
+#endif
 #include <fcntl.h>
 #include <stdlib.h>
 #include <string.h>
 #include <stdio.h>
 #include <setjmp.h>
 #include <signal.h>
+#ifndef _MSC_VER
 #include <sys/time.h>
+#endif
 #include <time.h>
 
 #if defined(__FreeBSD__)
diff -ur liboil-0.3.13/liboil/liboilparameter.h songbird/gst-sbird/trunk/dependencies/vendor/liboil/liboil/liboilparameter.h
--- liboil-0.3.13/liboil/liboilparameter.h	2008-02-22 11:59:17.000000000 -0800
+++ songbird/gst-sbird/trunk/dependencies/vendor/liboil/liboil/liboilparameter.h	2008-03-11 17:10:52.000000000 -0700
@@ -124,9 +124,9 @@
 
   OilArgType parameter_type;
 
-  void *src_data;
-  void *ref_data;
-  void *test_data;
+  uint8_t *src_data;
+  uint8_t *ref_data;
+  uint8_t *test_data;
   unsigned long value;
 
   int pre_n;
diff -ur liboil-0.3.13/liboil/liboilprofile.c songbird/gst-sbird/trunk/dependencies/vendor/liboil/liboil/liboilprofile.c
--- liboil-0.3.13/liboil/liboilprofile.c	2008-02-22 11:59:17.000000000 -0800
+++ songbird/gst-sbird/trunk/dependencies/vendor/liboil/liboil/liboilprofile.c	2008-03-11 17:10:52.000000000 -0700
@@ -32,7 +32,9 @@
 #include <liboil/liboilprofile.h>
 #include <liboil/liboildebug.h>
 
+#ifndef _MSC_VER
 #include <sys/time.h>
+#endif
 #include <time.h>
 #include <string.h>
 #include <math.h>
diff -ur liboil-0.3.13/liboil/liboiltest.c songbird/gst-sbird/trunk/dependencies/vendor/liboil/liboil/liboiltest.c
--- liboil-0.3.13/liboil/liboiltest.c	2008-02-22 11:59:17.000000000 -0800
+++ songbird/gst-sbird/trunk/dependencies/vendor/liboil/liboil/liboiltest.c	2008-03-11 17:10:52.000000000 -0700
@@ -671,7 +671,7 @@
   }
 
   for(i=0;i<post_n;i++){
-    if (!check_guard (data + stride * i + chunk_size, hole_size, guard)) {
+    if (!check_guard ((uint8_t *) data + stride * i + chunk_size, hole_size, guard)) {
       return 0;
     }
   }
--- liboil-0.3.13/liboil/mmx/recon8x8_mmx.c	2008-02-22 11:59:17.000000000 -0800
+++ songbird/gst-sbird/trunk/dependencies/vendor/liboil/liboil/mmx/recon8x8_mmx.c	2008-03-11 17:10:51.000000000 -0700
@@ -31,6 +31,7 @@
 #include <liboil/liboilfunction.h>
 #include <mmintrin.h>
 
+#ifdef ENABLE_BROKEN_IMPLS
 union m64_int {
   __m64 m64;
   uint64_t ull;
@@ -64,6 +65,7 @@
   _mm_empty();
 }
 OIL_DEFINE_IMPL_FULL (recon8x8_intra_mmx, recon8x8_intra, OIL_IMPL_FLAG_MMX);
+#endif
 
 static void
 recon8x8_inter_mmx (uint8_t *dest, int ds, uint8_t *src, int ss,
