--- liboil-0.3.13/examples/jpeg/jpeg.c	2008-02-22 11:59:17.000000000 -0800
+++ songbird/gst-sbird/trunk/dependencies/vendor/liboil/examples/jpeg/jpeg.c	2008-03-11 17:10:52.000000000 -0700
@@ -1,7 +1,9 @@
 
 #include <sys/stat.h>
 #include <fcntl.h>
+#ifdef HAVE_UNISTD_H
 #include <unistd.h>
+#endif
 #include <stdlib.h>
 #include <stdio.h>
 #include <string.h>
--- liboil-0.3.13/examples/jpeg/jpeg.h	2008-02-22 11:59:17.000000000 -0800
+++ songbird/gst-sbird/trunk/dependencies/vendor/liboil/examples/jpeg/jpeg.h	2008-03-11 17:10:52.000000000 -0700
@@ -2,7 +2,9 @@
 #ifndef _JPEG_DECODER_H_
 #define _JPEG_DECODER_H_
 
+#ifdef HAVE_STDINT_H
 #include <stdint.h>
+#endif
 
 #define JPEG_MARKER_STUFFED		0x00
 #define JPEG_MARKER_TEM			0x01
@@ -69,8 +71,7 @@
 	unsigned char **image, int *rowstride);
 
 unsigned char *jpeg_decoder_get_argb_image (JpegDecoder *dec);
-int jpeg_decode_argb (uint8_t *data, int length, uint32_t **image,
-    int *width, int *height);
+int jpeg_decode_argb (uint8_t *data, int length, uint32_t **image, int *width, int *height);
 
 #endif
 
--- liboil-0.3.13/examples/jpeg/jpeg_huffman.c	2008-02-22 11:59:17.000000000 -0800
+++ songbird/gst-sbird/trunk/dependencies/vendor/liboil/examples/jpeg/jpeg_huffman.c	2008-03-11 17:10:52.000000000 -0700
@@ -1,7 +1,9 @@
 
 #include <sys/stat.h>
 #include <fcntl.h>
+#ifdef HAVE_UNISTD_H
 #include <unistd.h>
+#endif
 #include <stdlib.h>
 #include <stdio.h>
 #include <string.h>
diff -ur liboil-0.3.13/examples/jpeg/test.c songbird/gst-sbird/trunk/dependencies/vendor/liboil/examples/jpeg/test.c
--- liboil-0.3.13/examples/jpeg/test.c	2008-02-22 11:59:17.000000000 -0800
+++ songbird/gst-sbird/trunk/dependencies/vendor/liboil/examples/jpeg/test.c	2008-03-11 17:10:52.000000000 -0700
@@ -1,7 +1,9 @@
 
 #include <sys/stat.h>
 #include <fcntl.h>
+#ifdef HAVE_UNISTD_H
 #include <unistd.h>
+#endif
 #include <stdlib.h>
 #include <stdio.h>
 
diff -ur liboil-0.3.13/examples/jpeg/test_rgb.c songbird/gst-sbird/trunk/dependencies/vendor/liboil/examples/jpeg/test_rgb.c
--- liboil-0.3.13/examples/jpeg/test_rgb.c	2008-02-22 11:59:17.000000000 -0800
+++ songbird/gst-sbird/trunk/dependencies/vendor/liboil/examples/jpeg/test_rgb.c	2008-03-11 17:10:52.000000000 -0700
@@ -1,7 +1,9 @@
 
 #include <sys/stat.h>
 #include <fcntl.h>
+#ifdef HAVE_UNISTD_H
 #include <unistd.h>
+#endif
 #include <stdlib.h>
 #include <stdio.h>
 
--- liboil-0.3.13/examples/md5/md5sum.c	2008-02-22 11:59:17.000000000 -0800
+++ songbird/gst-sbird/trunk/dependencies/vendor/liboil/examples/md5/md5sum.c	2008-03-11 17:10:52.000000000 -0700
@@ -41,7 +41,9 @@
 #endif
 #include <errno.h>
 #include <fcntl.h>
+#ifdef HAVE_UNISTD_H
 #include <unistd.h>
+#endif
 
 #ifndef MAP_POPULATE
 #define MAP_POPULATE 0
--- liboil-0.3.13/liboil/c/copy.c	2008-02-22 11:59:17.000000000 -0800
+++ songbird/gst-sbird/trunk/dependencies/vendor/liboil/liboil/c/copy.c	2008-03-11 17:10:52.000000000 -0700
@@ -33,7 +33,9 @@
 #include <liboil/liboilfunction.h>
 #include <liboil/liboilclasses.h>
 
+#ifdef HAVE_UNISTD_H
 #include <unistd.h>
+#endif
 
 
 static void
diff -ur liboil-0.3.13/liboil/c/swab.c songbird/gst-sbird/trunk/dependencies/vendor/liboil/liboil/c/swab.c
--- liboil-0.3.13/liboil/c/swab.c	2008-02-22 11:59:17.000000000 -0800
+++ songbird/gst-sbird/trunk/dependencies/vendor/liboil/liboil/c/swab.c	2008-03-11 17:10:52.000000000 -0700
@@ -33,7 +33,9 @@
 #include <liboil/liboilfunction.h>
 #include <liboil/liboilclasses.h>
 
+#ifdef HAVE_UNISTD_H
 #include <unistd.h>
+#endif
 #include <string.h>
 
 
--- liboil-0.3.13/liboil/liboilcpu-arm.c	2008-02-22 11:59:17.000000000 -0800
+++ songbird/gst-sbird/trunk/dependencies/vendor/liboil/liboil/liboilcpu-arm.c	2008-03-11 17:10:52.000000000 -0700
@@ -34,7 +34,9 @@
 #include <liboil/liboilfault.h>
 #include <liboil/liboilutils.h>
 
+#ifdef HAVE_UNISTD_H
 #include <unistd.h>
+#endif
 #include <fcntl.h>
 #include <stdlib.h>
 #include <string.h>
diff -ur liboil-0.3.13/liboil/liboilcpu-misc.c songbird/gst-sbird/trunk/dependencies/vendor/liboil/liboil/liboilcpu-misc.c
--- liboil-0.3.13/liboil/liboilcpu-misc.c	2008-02-22 11:59:17.000000000 -0800
+++ songbird/gst-sbird/trunk/dependencies/vendor/liboil/liboil/liboilcpu-misc.c	2008-03-11 17:10:52.000000000 -0700
@@ -34,7 +34,9 @@
 #include <liboil/liboilfault.h>
 #include <liboil/liboilutils.h>
 
+#ifdef HAVE_UNISTD_H
 #include <unistd.h>
+#endif
 #include <fcntl.h>
 #include <stdlib.h>
 #include <string.h>
--- liboil-0.3.13/liboil/liboilutils.c	2008-02-22 11:59:17.000000000 -0800
+++ songbird/gst-sbird/trunk/dependencies/vendor/liboil/liboil/liboilutils.c	2008-03-11 17:10:52.000000000 -0700
@@ -32,7 +32,9 @@
 #include <liboil/liboildebug.h>
 #include <liboil/liboilutils.h>
 
+#ifdef HAVE_UNISTD_H
 #include <unistd.h>
+#endif
 #include <fcntl.h>
 #include <stdlib.h>
 #include <string.h>
