--- liboil-0.3.13/m4/ax_create_stdint_h.m4	2008-02-22 11:59:17.000000000 -0800
+++ songbird/gst-sbird/trunk/dependencies/vendor/liboil/m4/ax_create_stdint_h.m4	2008-03-11 17:10:50.000000000 -0700
@@ -519,7 +519,7 @@
 #define __intptr_t_defined
 /* we encourage using "long" to store pointer values, never use "int" ! */
 #if   _STDINT_LONG_MODEL+0 == 242 || _STDINT_LONG_MODEL+0 == 484
-typedef  unsinged int   uintptr_t;
+typedef  unsigned int   uintptr_t;
 typedef           int    intptr_t;
 #elif _STDINT_LONG_MODEL+0 == 244 || _STDINT_LONG_MODEL+0 == 444
 typedef  unsigned long  uintptr_t;
